kubernetes-manifests/emailservice.yaml

```diff
@@ -26,6 +26,10 @@ spec:
         app: emailservice
     spec:
       terminationGracePeriodSeconds: 5
+      initContainers:
+      - name: wait-for-gke-metadata-server
+        image: gcr.io/stackdriver-sandbox-230822/sandbox/emailservice:latest
+        command: ['sh', '-c', 'sleep 5']
```

Comment:
Can we try to reach metadata endpoint instead of waiting 5 sec?

Reply:
Trying it with `for i in {1..10}; do sleep 1; if nslookup metadata.google.internal; then exit 0; fi; done` seems to work. It doesn't break local runs/CI but it introduces a some latency.

In principle, there could still be x% chance that metadata server takes longer than 10s... we can always increase the timeout duration.